onnx/backend/test/runner/__init__.py

```diff
@@ -317,18 +365,62 @@ def run(test_self: Any, device: str) -> None:
                 and not self.backend.is_compatible(model)
             ):
                 raise unittest.SkipTest("Not compatible with backend")
+
             prepared_model = self.backend.prepare(model, device)
             assert prepared_model is not None
 
-            # TODO after converting all npz files to protobuf, we can delete this.
-            for test_data_npz in glob.glob(os.path.join(model_dir, "test_data_*.npz")):
-                test_data = np.load(test_data_npz, encoding="bytes")
-                inputs = list(test_data["inputs"])
+            if use_dummy:
+                # We generate dummy data and compute the expected output
+                # with the reference runtime.
+                with open(model_pb_path, "rb") as f:
+                    onx = onnx.load(f)
+                test_data_set = os.path.join(model_dir, "test_data_set_0")
+                if not os.path.exists(test_data_set):
+                    os.mkdir(test_data_set)
+                ref = onnx.reference.ReferenceEvaluator(onx, verbose=2)
+                feeds = {}
+                inits = set(i.name for i in onx.graph.initializer)
+                n_input = 0
+                inputs = []
+                for i in range(len(onx.graph.input)):
+                    if onx.graph.input[i].name in inits:
+                        continue
+                    name = os.path.join(test_data_set, f"input_{n_input}.pb")
+                    inputs.append(name)
+                    n_input += 1
+                    x = onx.graph.input[i]
+                    value = self.generate_random_data(
+                        x, seed=0, name=model_test.model_name
+                    )
+                    feeds[x.name] = value
+                    with open(name, "wb") as f:
+                        f.write(onnx.numpy_helper.from_array(value).SerializeToString())
+                # Reference implementation of operator Conv is too slow.
```

Comment:
Don't understand above comment. Is that some copy-paste error?

Reply:
It is not from a copy paste. ReferenceEvaluator is used to generate the expected output of the light models. It is slow if one of them is including a Conv operator which is usually the case. I extended the comment and also create a new test file checking the backend API with the ReferenceEvaluator.